nifi-nar-bundles/nifi-airtable-bundle/nifi-airtable-processors/src/main/java/org/apache/nifi/processors/airtable/service/AirtableRestService.java

```diff
@@ -0,0 +1,123 @@
+        this.webClientServiceProvider = webClientServiceProvider;
+        this.apiUrl = apiUrl;
+        this.apiKey = apiKey;
+        this.baseId = baseId;
+        this.tableId = tableId;
+    }
+
+    public <R> R getRecords(final AirtableGetRecordsParameters getRecordsParameters, final Function<InputStream, R> callback) {
+        final URI uri = buildUri(getRecordsParameters);
+        try (final HttpResponseEntity response = webClientServiceProvider.getWebClientService()
+                .get()
+                .uri(uri)
+                .header("Authorization", "Bearer " + apiKey)
+                .retrieve()) {
+
+            final InputStream bodyInputStream = response.body();
+            if (SUCCESSFUL_RESPONSE_RANGE.contains(response.statusCode())) {
+                return callback.apply(bodyInputStream);
+            }
+            if (response.statusCode() == TOO_MANY_REQUESTS) {
+                throw new RateLimitExceededException();
+            }
+            final StringBuilder exceptionMessageBuilder = new StringBuilder("Error response. Code: " + response.statusCode());
+            final String bodyText = IOUtils.toString(bodyInputStream, StandardCharsets.UTF_8);
+            if (bodyText != null) {
+                exceptionMessageBuilder.append(" Body: ").append(bodyText);
+            }
+
+            throw new ProcessException(exceptionMessageBuilder.toString());
+        } catch (IOException e) {
+            throw new RuntimeException(String.format("Airtable HTTP request failed [%s]", uri), e);
+        }
+    }
+
+    public HttpUriBuilder createUriBuilder(final boolean includePort) {
+        final URI uri = URI.create(apiUrl);
+        final HttpUriBuilder uriBuilder = webClientServiceProvider.getHttpUriBuilder()
+                .scheme(uri.getScheme())
+                .host(uri.getHost())
+                .encodedPath(uri.getPath())
+                .addPathSegment(baseId)
+                .addPathSegment(tableId);
+        if (includePort && uri.getPort() != -1) {
+            uriBuilder.port(uri.getPort());
+        }
+        return uriBuilder;
+    }
+
+    private URI buildUri(AirtableGetRecordsParameters getRecordsParameters) {
+        final HttpUriBuilder uriBuilder = createUriBuilder(true);
+        for (final String field : getRecordsParameters.getFields()) {
+            uriBuilder.addQueryParameter("fields[]", field);
+        }
+
+        final List<String> filters = new ArrayList<>();
+        getRecordsParameters.getCustomFilter()
+                .ifPresent(filters::add);
+        getRecordsParameters.getModifiedAfter()
+                .map(modifiedAfter -> "IS_AFTER(LAST_MODIFIED_TIME(),DATETIME_PARSE(\"" + modifiedAfter + "\"))")
+                .ifPresent(filters::add);
+        getRecordsParameters.getModifiedBefore()
+                .map(modifiedBefore -> "IS_BEFORE(LAST_MODIFIED_TIME(),DATETIME_PARSE(\"" + modifiedBefore + "\"))")
+                .ifPresent(filters::add);
```

Comment:
As far as I can see (please double check it), both `IS_AFTER` and `IS_BEFORE` are exclusive but one endpoint of the query time window must be inclusive in order not to loose records with Last Modified time equal to the time window start/end. The start point should be inclusive.